test/e2e/common/expansion.go

```diff
@@ -306,6 +309,325 @@ var _ = framework.KubeDescribe("Variable Expansion", func() {
+						Env: []v1.EnvVar{
+							{
+								Name:  "POD_NAME",
+								Value: "foo",
+							},
+							{
+								Name: "ANNOTATION",
+								ValueFrom: &v1.EnvVarSource{
+									FieldRef: &v1.ObjectFieldSelector{
+										APIVersion: "v1",
+										FieldPath:  "metadata.annotations['mysubpath']",
+									},
+								},
+							},
+						},
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:        "workdir1",
+								MountPath:   "/subpath_mount",
+								SubPathExpr: "$(ANNOTATION)/$(POD_NAME)",
+							},
+							{
+								Name:      "workdir2",
+								MountPath: "/volume_mount",
+							},
+						},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+				Volumes: []v1.Volume{
+					{
+						Name: "workdir1",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{Path: "/tmp"},
+						},
+					},
+					{
+						Name: "workdir2",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{Path: "/tmp"},
+						},
+					},
+				},
+			},
+		}
+
+		By("creating the pod")
+		pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
+
+		By("waiting for pod running")
+		err = framework.WaitTimeoutForPodRunningInNamespace(f.ClientSet, pod.Name, pod.Namespace, framework.PodStartShortTimeout)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for pod to be running")
+
+		By("creating a file in subpath")
+		cmd := "touch /volume_mount/mypath/foo/test.log"
+		_, err = framework.RunHostCmd(pod.Namespace, pod.Name, cmd)
+		if err != nil {
+			framework.Failf("expected to be able to write to subpath")
+		}
+
+		By("test for file in mounted path")
+		cmd = "test -f /subpath_mount/test.log"
+		_, err = framework.RunHostCmd(pod.Namespace, pod.Name, cmd)
+		if err != nil {
+			framework.Failf("expected to be able to verify file")
+		}
+
+		By("updating the annotation value")
+		var podClient *framework.PodClient
+		podClient = f.PodClient()
+
+		podClient.Update(podName, func(pod *v1.Pod) {
+			pod.ObjectMeta.Annotations["mysubpath"] = "mynewpath"
+		})
+
+		By("waiting for annotated pod running")
+		err = framework.WaitTimeoutForPodRunningInNamespace(f.ClientSet, pod.Name, pod.Namespace, framework.PodStartShortTimeout)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for annotated pod to be running")
+
+		By("deleting the pod gracefully")
+		err = framework.DeletePodWithWait(f, f.ClientSet, pod)
+		Expect(err).NotTo(HaveOccurred(), "failed to delete pod")
+	})
+
+	/*
+		    Testname: var-expansion-subpath-lifecycle
+		    Description: Verify that a subpath expansion can be modified during the lifecycle of a container.
+			1.	valid subpathexpr starts a container running
+			2.	test for valid subpath writes
+			3.	container restarts
+			4.  delete cleanly
+
+	*/
+	It("should succeed in altering lifecycle of container [Feature:VolumeSubpathEnvExpansion][NodeAlphaFeature:VolumeSubpathEnvExpansion][Slow]", func() {
```

Comment:
maybe a more descriptive name would be "should not change the subpath mount on a container restart if the environment variable changes"

Reply:
ah sorry to be nitpicky. Could you also change the name of the test case in the `It()` to also match the description?

Reply:
Done